HeterogeneousCore/SonicTriton/src/TritonService.cc

```diff
@@ -297,20 +300,39 @@ void TritonService::preBeginJob(edm::PathsAndConsumesOfModulesBase const&, edm::
                                            << output;
 }
 
+void TritonService::notifyCallStatus(bool status) const {
+  if (status)
+    --callFails_;
+  else
+    ++callFails_;
+}
+
 void TritonService::postEndJob() {
   if (!startedFallback_)
     return;
 
-  std::string command = fallbackOpts_.command + " stop";
+  std::string command = fallbackOpts_.command;
+  //prevent log cleanup during server stop
+  if (callFails_ > 0)
+    command += " -c";
+  command += " stop";
   if (verbose_)
     edm::LogInfo("TritonService") << command;
 
   const auto& [output, rv] = execSys(command);
-  if (rv != 0) {
+  if (rv != 0 or callFails_ > 0) {
+    //print logs if cmsRun is currently exiting because of a TritonException
     edm::LogError("TritonService") << output;
     printFallbackServerLog<edm::LogError>();
-    throw cms::Exception("FallbackFailed")
-        << "TritonService: Stopping the fallback server failed with exit code " << rv;
+    if (rv != 0) {
+      cms::Exception stopException("FallbackFailed");
+      stopException << "TritonService: Stopping the fallback server failed with exit code " << rv;
```

Comment:
I'd suggest to construct a string here that is used in the `LogWarning` and in the exception throwing, rather than using `cms::Exception` as a "string stream" for the `LogWarning`.

Reply:
Done